src/project/mod.rs

```diff
@@ -541,6 +567,31 @@ impl Project {
         Ok((name, nameless))
     }
 
+    fn add_pypi_dep_to_table(
+        deps_table: &mut Item,
+        name: &rip::types::PackageName,
+        requirement: &PyPiRequirement,
+    ) -> miette::Result<()> {
+        // If it doesn't exist create a proper table
+        if deps_table.is_none() {
+            *deps_table = Item::Table(Table::new());
+        }
+
+        // Cast the item into a table
+        let deps_table = deps_table.as_table_like_mut().ok_or_else(|| {
+            miette::miette!("dependencies in {} are malformed", consts::PROJECT_MANIFEST)
+        })?;
+
+        let requirement_string = if let Some(version) = requirement.clone().version {
+            version.to_string()
+        } else {
+            "*".to_string()
+        };
+        // Store (or replace) in the document
+        deps_table.insert(name.as_str(), Item::Value(requirement_string.into()));
```

Comment:
Extras are not serialized and maybe you can create a `From<PyPiRequirement> for Item` (or `Into<Item> for PyPiRequirement`) if the compiler complains.

Reply:
Added the `From` functionality